elephant/spectral.py

```diff
@@ -522,6 +525,359 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
+
+        temp = (spectrum_estimates[np.newaxis, :, :, :]
+                * np.conjugate(spectrum_estimates[:, np.newaxis, :, :]))
+
+        # Average Fourier transform windowed signal
+        cross_spec_estimates[i] = np.mean(temp,
+                                          axis=-2,
+                                          dtype=np.complex64) / fs
+
+        phase_spec_estimates[i] = np.angle(np.mean(temp, axis=-2))
+
+    cross_spec = np.mean(cross_spec_estimates, axis=0)
+    phase_cross_spec = np.mean(phase_spec_estimates, axis=0)
+
+    # Attach proper units to return values
+    if isinstance(signals, pq.quantity.Quantity):
+        cross_spec = cross_spec * signals.units * signals.units / pq.Hz
+        phase_cross_spec = phase_cross_spec * pq.rad
+        freqs = freqs * pq.Hz
+
+    return freqs, phase_cross_spec, cross_spec
+
+
+def multitaper_coherence(signal_i, signal_j, n_segments=8, len_segment=None,
+                         frequency_resolution=None, overlap=0.5, fs=1,
+                         nw=4, num_tapers=None, peak_resolution=None):
+    """
+    Estimates the magnitude-squared coherence and phase-lag of two given
+    'neo.AnalogSignal' using the Multitaper method.
+    The function calls `multitaper_cross_spectrum` internally and thus both
+    functions share parts of the respective signatures.
+
+    .. math::
+        C(\omega)=\frac{|S_{xy}(\omega)|^2}{S_{xx}(\omega)S_{yy}(\omega)}
+
+    Parameters
+    ----------
+    signal_i : neo.AnalogSignal or pq.Quantity or np.ndarray
+        First time series data of the pair between which coherence is
+        computed.
+    signal_j : neo.AnalogSignal or pq.Quantity or np.ndarray
+        Second time series data of the pair between which coherence is
+        computed.
+        The shapes and the sampling frequencies of `signal_i` and `signal_j`
+        must be identical. When `signal_i` and `signal_j` are not
+        `neo.AnalogSignal`, sampling frequency should be specified through the
+        keyword argument `fs`. Otherwise, the default value is used
+        (`fs` = 1.0).
+    n_segments : int, optional
+        Number of segments. The length of segments is adjusted so that
+        overlapping segments cover the entire stretch of the given data. This
+        parameter is ignored if `len_segment` or `frequency_resolution` is
+        given.
+        Default: 8.
+    len_segment : int, optional
+        Length of segments. This parameter is ignored if `frequency_resolution`
+        is given. If None, it will be determined from other parameters.
+        Default: None.
+    frequency_resolution : pq.Quantity or float, optional
+        Desired frequency resolution of the obtained PSD estimate in terms of
+        the interval between adjacent frequency bins. When given as a `float`,
+        it is taken as frequency in Hz.
+        If None, it will be determined from other parameters.
+        Default: None.
```

Comment:
```suggestion
```

remove (?), see comment above.

Reply:
Fixed in https://github.com/NeuralEnsemble/elephant/pull/525/commits/1b3f316afa1cd246ea4cd6efdda6c3b09fa8f9e1